src/Compilation.zig

```diff
@@ -1459,16 +1464,14 @@ pub fn performAllTheWork(self: *Compilation) error{ TimerUnsupported, OutOfMemor
                 }
 
                 assert(decl.typed_value.most_recent.typed_value.ty.hasCodeGenBits());
-
                 self.bin_file.updateDecl(module, decl) catch |err| {
                     switch (err) {
                         error.OutOfMemory => return error.OutOfMemory,
                         error.AnalysisFail => {
                             decl.analysis = .dependency_failure;
                         },
                         else => {
-                            try module.failed_decls.ensureCapacity(module.gpa, module.failed_decls.items().len + 1);
-                            module.failed_decls.putAssumeCapacityNoClobber(decl, try ErrorMsg.create(
+                            try module.addDeclErr(decl, try ErrorMsg.create(
```

Comment:
Every instance of this change introduced a memory leak if `ErrorMsg.create` fails.